contract/test/wallet-tools.js

```diff
@@ -0,0 +1,227 @@
+/**
+ * @file smart-wallet test tools
+ *
+ * @see {mockWalletFactory}
+ */
+// @ts-check
+import { E, Far } from '@endo/far';
+import { makePromiseKit } from '@endo/promise-kit';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { makeNotifier } from '@agoric/notifier';
+
+import { allValues, mapValues } from '../src/objectTools.js';
+
+/**
+ * @typedef {import('@agoric/smart-wallet/src/offers.js').OfferSpec} OfferSpec
+ * @typedef {import('@agoric/smart-wallet/src/smartWallet.js').UpdateRecord} UpdateRecord
+ */
+
+const { values } = Object;
+const { Fail, quote: q } = assert;
+
+const DEPOSIT_FACET_KEY = 'depositFacet'; // XXX does agoric-sdk export this?
+
+/**
+ * @param {{
+ *   zoe: ERef<ZoeService>;
+ *   namesByAddressAdmin: ERef<import('@agoric/vats').NameAdmin>;
+ * }} powers
+ * @param {IssuerKeywordRecord} issuerKeywordRecord
+ *
+ * @typedef {Awaited<ReturnType<Awaited<ReturnType<typeof mockWalletFactory>['makeSmartWallet']>>>} MockWallet
+ */
+export const mockWalletFactory = (
+  { zoe, namesByAddressAdmin },
+  issuerKeywordRecord,
+) => {
+  /** @param {string} address */
+  const makeSmartWallet = async address => {
+    const { nameAdmin: addressAdmin } = await E(
+      namesByAddressAdmin,
+    ).provideChild(address, [DEPOSIT_FACET_KEY]);
+
+    const entries = await Promise.all(
+      values(issuerKeywordRecord).map(async issuer => {
+        const purse = await E(issuer).makeEmptyPurse();
+        const brand = await E(issuer).getBrand();
+        /** @type {[Brand, Purse]} */
+        const entry = [brand, purse];
+        return entry;
+      }),
+    );
+    const purseByBrand = new Map(entries);
+    const invitationBrand = await E(E(zoe).getInvitationIssuer()).getBrand();
+    purseByBrand.has(invitationBrand) ||
+      Fail`no invitation issuer / purse / brand`;
+    const invitationPurse = purseByBrand.get(invitationBrand);
+    assert(invitationPurse);
+
+    const depositFacet = Far('DepositFacet', {
+      /** @param {Payment} pmt */
+      receive: async pmt => {
+        const pBrand = await E(pmt).getAllegedBrand();
+        if (!purseByBrand.has(pBrand))
+          throw Error(`brand not known/supported: ${pBrand}`);
+        const purse = purseByBrand.get(pBrand);
+        assert(purse);
+        return E(purse).deposit(pmt);
+      },
+    });
+    await E(addressAdmin).default(DEPOSIT_FACET_KEY, depositFacet);
+
+    const getContractInvitation = invitationSpec => {
+      const {
+        instance,
+        publicInvitationMaker,
+        invitationArgs = [],
+      } = invitationSpec;
+      const pf = E(zoe).getPublicFacet(instance);
+      return E(pf)[publicInvitationMaker](...invitationArgs);
+    };
+
+    const getPurseInvitation = async invitationSpec => {
+      //   const { instance, description } = invitationSpec;
```

Comment:
is this line obsolete?

Reply:
it's sort of part of the todo... which I have just todone